include/Dialect/Openfhe/IR/OpenfheTypes.td

```diff
@@ -0,0 +1,55 @@
+#ifndef INCLUDE_DIALECT_OPENFHE_IR_OPENFHETYPES_TD_
+#define INCLUDE_DIALECT_OPENFHE_IR_OPENFHETYPES_TD_
+
+include "OpenfheDialect.td"
+
+include "mlir/IR/AttrTypeBase.td"
+include "mlir/IR/CommonTypeConstraints.td"
+include "mlir/IR/DialectBase.td"
+include "mlir/IR/OpBase.td"
+include "mlir/Interfaces/InferTypeOpInterface.td"
+
+class Openfhe_Type<string name, string typeMnemonic, list<Trait> traits = []>
+    : TypeDef<Openfhe_Dialect, name, traits> {
+  let mnemonic = typeMnemonic;
+}
+
+// Plaintext
+class Openfhe_Plaintext<string scheme>
+  : Openfhe_Type< "Plaintext" # scheme, "plaintext" # scheme> {
+  let summary = "A plaintext corresponding to OpenFHE's ciphertext type in " # scheme # " scheme";
+}
+
+foreach scheme = ["BFV", "BGV", "CKKS"] in {
+  def Openfhe_Plaintext # scheme : Openfhe_Plaintext<scheme>;
+}
+
+def Openfhe_PlaintextType :
+  AnyTypeOf<[
```

Comment:
After looking over the OpenFHE pke API, it does appear that they have a [single plaintext type](https://github.com/openfheorg/openfhe-development/blob/4ebb28ea7bdd894a73bc5b73e59fcfbc78253308/src/pke/include/encoding/plaintext-fwd.h#L44 and [ciphertext type](https://github.com/openfheorg/openfhe-development/blob/4ebb28ea7bdd894a73bc5b73e59fcfbc78253308/src/pke/include/ciphertext-fwd.h#L45) for all three arithmetic schemes, and their ciphertext type is (in every example I can see from their docs) using the "double-CRT" representation. I think it would be a reasonable starting point to reuse the `lwe` types here, and effectively hard-code the double-CRT encoding choice when generating code. We can generalize it later as needed.

Reply:
Agree to put rlwe ciphertext types from the `lwe` dialect - but we currently don't have that implemented (we only have lwe ciphertexts there). So it may be worth putting the unified ciphertexts types there and then add their attributes (https://github.com/openfheorg/openfhe-development/blob/4ebb28ea7bdd894a73bc5b73e59fcfbc78253308/src/pke/include/ciphertext.h#L101) there when needed

Reply:
Do you mean that we need to add new `RLWECiphertextType` in the `lwe` dialect to represent the "unified ciphertexts types?

Reply:
I think so! Lattigo also has a unified ciphertext type so it would also use a generic `RLWECiphertextType` (https://pkg.go.dev/github.com/ldsec/lattigo/v2/rlwe)

Reply:
Applied!